woocommerce/woocommerce
# Fix and consolidate linting across the monorepo

### All Submissions:






### Changes proposed in this Pull Request:

Linting was broken across the monorepo after recent changes to pnpm version. There were are number of reasons for breakages, such as some dependencies not being declared in packages, or misconfiguration.

To fix this as simply as possible, I've migrated every package across to a consistent ts lint version and I've made sure every package has our eslint plugin as a dependency. This meant I had to fix some eslint issues. I used auto fixing where possible, I added some ignores in the `js/api` package that someone can fix later, and I did manual fixes myself where it seemed quick and worthwhile to do so.


### How to test the changes in this Pull Request:

1. See the CI lint job pass
2. Lint locally. To do so I would clear pnpm cache first `pnpm store prune`, then install deps `pnpm i`
3. Then run lint from the root dir: `pnpm -r run lint`



### FOR PR REVIEWER ONLY:
